### PR TITLE
POR-2041: add defer func to update that cancels builds if workflow is canceled; handle cancel on front end

### PR DESCRIPTION
**Pull request by d-g-town**
Screenshot 2023-11-07 at 12 44 13 PM
Screenshot 2023-11-07 at 12 44 32 PM

### CODE REVIEW

File: dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/BuildEventCard.tsx [hunk 1]
```diff
@@ -45,40 +45,47 @@ const BuildEventCard: React.FC<Props> = ({
         {match(event.status)
           .with("SUCCESS", () => "Build successful")
           .with("FAILED", () => "Build failed")
+          .with("CANCELED", () => "Build cancelled")
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/BuildEventCard.tsx:48]: cancelled -> canceled

File: dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/BuildEventCard.tsx [hunk 2]
```diff
@@ -45,40 +45,47 @@ const BuildEventCard: React.FC<Props> = ({
...
+        </Wrapper>
+    );
+  }
+
   const renderInfoCta = (event: PorterAppBuildEvent) => {
     switch (event.status) {
       case "SUCCESS":
         return null;
+      case "CANCELED":
+        return renderLogsAndRetry(event);
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/BuildEventCard.tsx:86]:
```suggestion
      case "CANCELED":
      case "FAILED": 
        return renderLogsAndRetry(event);
```

File: dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/focus-views/BuildFailureEventFocusView.tsx [hunk 3]
```diff
@@ -3,15 +3,15 @@ import Spacer from "components/porter/Spacer";
 import React, { useEffect, useRef, useState } from "react";
 import api from "shared/api";
 import styled from "styled-components";
-import Anser, { AnserJsonEntry } from "anser";
+import Anser, { type AnserJsonEntry } from "anser";
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/focus-views/BuildFailureEventFocusView.tsx:6]: please rename this file

File: cli/cmd/v2/update.go [hunk 4]
```diff
@@ -35,6 +37,20 @@ type UpdateInput struct {
...
 func Update(ctx context.Context, inp UpdateInput) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	go func() {
+		termChan := make(chan os.Signal, 1)
+		signal.Notify(termChan, syscall.SIGINT, syscall.SIGTERM)
+		select {
+		case <-termChan:
+			color.New(color.FgYellow).Printf("Shutdown signal received, cancelling processes\n") // nolint:errcheck,gosec
```
Comment (ferozemohideen) [inline cli/cmd/v2/update.go:48]: canceling

File: cli/cmd/commands/errors.go [hunk 5]
```diff
@@ -79,6 +79,11 @@ func checkLoginAndRunWithConfig(cmd *cobra.Command, cliConf config.CLIConfig, ar
 			return nil
 		}
 
+		if errors.Is(err, context.Canceled) {
+			color.New(color.FgYellow).Println("Command was cancelled") // nolint:errcheck,gosec
```
Comment (ferozemohideen) [inline cli/cmd/commands/errors.go:83]: canceled

**Review by ferozemohideen — APPROVED** [review]
nice

File: dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/BuildEventCard.tsx (hunk 2 above)
Comment (d-g-town) [reply dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/BuildEventCard.tsx:86]: `go` does not allow this kind of logic, and I think for consistency I'd prefer to explicitly make the call and return

**Pull request merged**